chia/wallet/wallet_state_manager.py

```diff
@@ -529,13 +528,39 @@ async def synced(self):
             return True
         return False
 
-    def set_sync_mode(self, mode: bool, sync_height: uint32 = uint32(0)):
-        """
-        Sets the sync mode. This changes the behavior of the wallet node.
-        """
-        self.sync_mode = mode
-        self.sync_target = sync_height
-        self.state_changed("sync_changed")
+    @property
+    def sync_mode(self) -> bool:
+        return self._sync_target is not None
+
+    @property
+    def sync_target(self) -> Optional[uint32]:
+        return self._sync_target
+
+    @asynccontextmanager
+    async def set_sync_mode(self, target_height: uint32) -> AsyncIterator[uint32]:
+        if self.log.level == logging.DEBUG:
+            self.log.debug(f"set_sync_mode enter {await self.blockchain.get_finished_sync_up_to()}-{target_height}")
```

Comment:
given that you log this before we acquire the lock, the actual start height may vary, right?
It seems like it would make sense to log this after the acquire the lock, and also only call `get_finished_sync_up_to()` just once, instead of twice.

Reply:
The idea was to log the height before we try to acquire the lock and then again once we got it to see in the debug logs if there was something happening in between.